roxie/roxiemem/roxiemem.cpp

```diff
@@ -292,8 +292,19 @@ static void initializeHeap(bool allowHugePages, bool allowTransparentHugePages,
     if (!heapBase)
     {
         const memsize_t hugePageSize = getHugePageSize();
-        bool useTransparentHugePages = allowTransparentHugePages && areTransparentHugePagesEnabled();
+        HugePageMode mode = queryTransparentHugePagesMode();
+        bool hasTransparenHugePages = areTransparentHugePagesEnabled(mode);
+        bool useTransparentHugePages = allowTransparentHugePages && hasTransparenHugePages;
         memsize_t heapAlignment = useTransparentHugePages ? hugePageSize : HEAP_ALIGNMENT_SIZE;
+        if (mode == HugePageMode::Always)
+        {
+            //Always return memory in multiples of the huge page size - even if it is not being used
+            heapAlignment = hugePageSize;
+            OERRLOG("WARNING: The OS is configured to always use transparent huge pages.  This may cause unexplained pauses "
```

Comment:
Three separate strings, is this ok?

Reply:
ok, 3 strings format is ok - but I dont see it in my syslog msgs.  Let me debug.

Reply:
ok, so in Thorslave for example, is seems UseSysLogForOperatorMessages() is not called and so OERRLOG() does not show up in syslog.  Should we add
```
        if (globals->getPropBool("@enableSysLog",true))
            UseSysLogForOperatorMessages();
```
to Thorslave in thslavemain.cpp in or after startSlaveLog() ?

Reply:
they get concatenated

Reply:
hm, yes slaves should have on too (I thought did).

Yes, I think should add, but separate issue.